### PR TITLE
Recipe RemovedStateManagerMethods in jakarta-ee-10

### PR DESCRIPTION
**Pull request by ranuradh**
## What's changed?
image

## Anyone you would like to review specifically?
@joanvr 

### Checklist
- [X] I've added unit tests to cover both positive and negative cases
- [X] I've added the license header to any new files through `./gradlew licenseFormat`
- [X] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Review by joanvr — APPROVED** [review]

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml [hunk 1]
```diff
@@ -72,6 +73,42 @@ recipeList:
...
+name: org.openrewrite.java.migrate.jakarta.RemovedStateManagerMethods
+displayName: Use `StateManagementStrategy`
+description: >
+      Methods that were removed from the jakarta.faces.application.StateManager and javax.faces.application.StateManager classes in Jakarta Faces 4.0 are replaced
+      by jakarta.faces.view.StateManagementStrategy or javax.faces.view.StateManagementStrategy based on Jakarta10 migration in Faces 4.0
+recipeList:
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: "*.faces.application.StateManager getComponentStateToSave(*.faces.context.FacesContext)"
+      newMethodName: saveView
+      ignoreDefinition: false
```
Comment (cjobinabo) [inline src/main/resources/META-INF/rewrite/jakarta-ee-10.yml:87]: `ignoreDefinition` should be true.

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml [hunk 2]
```diff
@@ -72,6 +73,42 @@ recipeList:
...
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: "*.faces.application.StateManager getTreeStructureToSave(*.faces.context.FacesContext)"
+      newMethodName: saveView
+      ignoreDefinition: false
```
Comment (cjobinabo) [inline src/main/resources/META-INF/rewrite/jakarta-ee-10.yml:91]: `ignoreDefinition` should be true.

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml [hunk 3]
```diff
@@ -72,6 +73,42 @@ recipeList:
...
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: "*.faces.application.StateManager restoreComponentState(*.faces.context.FacesContext,*.faces.component.UIViewRoot,String)"
+      newMethodName: restoreView
+      ignoreDefinition: false
```
Comment (cjobinabo) [inline src/main/resources/META-INF/rewrite/jakarta-ee-10.yml:95]: `ignoreDefinition` should be true.

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml [hunk 4]
```diff
@@ -72,6 +73,42 @@ recipeList:
...
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: "*.faces.application.StateManager restoreTreeStructure(*.faces.context.FacesContext,String,String)"
+      newMethodName: restoreView
+      ignoreDefinition: false
```
Comment (cjobinabo) [inline src/main/resources/META-INF/rewrite/jakarta-ee-10.yml:99]: `ignoreDefinition` should be true.

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml [hunk 5]
```diff
@@ -72,6 +73,42 @@ recipeList:
...
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: "*.faces.application.StateManager saveSerializedView(*.faces.context.FacesContext)"
+      newMethodName: saveView
+      ignoreDefinition: false
```
Comment (cjobinabo) [inline src/main/resources/META-INF/rewrite/jakarta-ee-10.yml:103]: `ignoreDefinition` should be true.

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml (hunk 1 above)
Comment (ranuradh) [reply src/main/resources/META-INF/rewrite/jakarta-ee-10.yml]: updated it in the next commit

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml (hunk 2 above)
Comment (ranuradh) [reply src/main/resources/META-INF/rewrite/jakarta-ee-10.yml]: updated it in the next commit

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml (hunk 3 above)
Comment (ranuradh) [reply src/main/resources/META-INF/rewrite/jakarta-ee-10.yml]: updated it in the next commit

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml (hunk 5 above)
Comment (ranuradh) [reply src/main/resources/META-INF/rewrite/jakarta-ee-10.yml]: updated it in the next commit

**Review by cjobinabo — APPROVED** [review]

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml [hunk 6]
```diff
@@ -63,15 +64,51 @@ recipeList:
...
         oldFullyQualifiedTypeName: jakarta.xml.soap.SOAPElementFactory
         newFullyQualifiedTypeName: jakarta.xml.soap.SOAPFactory
         ignoreDefinition: true
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.jakarta.RemovedStateManagerMethods
+displayName: Use `StateManagementStrategy`
+description: >
+      Methods that were removed from the jakarta.faces.application.StateManager and javax.faces.application.StateManager classes in Jakarta Faces 4.0 are replaced
+      by jakarta.faces.view.StateManagementStrategy or javax.faces.view.StateManagementStrategy based on Jakarta10 migration in Faces 4.0
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/jakarta-ee-10.yml:82]: Would you mind adding back ticks around code references here as well, similar to the display name? That helps the docs look best if class and package names are rendered as code.

Comment (timtebeek) [reply src/main/resources/META-INF/rewrite/jakarta-ee-10.yml:82]:
```suggestion
      by `jakarta.faces.view.StateManagementStrategy` or `javax.faces.view.StateManagementStrategy` based on Jakarta10 migration in Faces 4.0
```

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml [hunk 7]
```diff
@@ -63,15 +64,51 @@ recipeList:
...
     - org.openrewrite.java.ChangeType:
         oldFullyQualifiedTypeName: jakarta.xml.soap.SOAPElementFactory
         newFullyQualifiedTypeName: jakarta.xml.soap.SOAPFactory
         ignoreDefinition: true
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.jakarta.RemovedStateManagerMethods
+displayName: Use `StateManagementStrategy`
+description: >
+      Methods that were removed from the jakarta.faces.application.StateManager and javax.faces.application.StateManager classes in Jakarta Faces 4.0 are replaced
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/jakarta-ee-10.yml:81]:
```suggestion
      Methods that were removed from the `jakarta.faces.application.StateManager` and `javax.faces.application.StateManager` classes in Jakarta Faces 4.0 are replaced
```

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml (hunk 6 above)
Comment (ranuradh) [reply src/main/resources/META-INF/rewrite/jakarta-ee-10.yml]: added in the next commit

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml (hunk 7 above)
Comment (ranuradh) [reply src/main/resources/META-INF/rewrite/jakarta-ee-10.yml]: added in the next commit

**Review by timtebeek — APPROVED** [review]

**Comment by timtebeek** [comment]
Thanks for the new recipes and follow up on the feedback received!

**Comment by ranuradh** [comment]
@timtebeek could you merge it? Thanks!

**Comment by timtebeek** [comment]
> @timtebeek could you merge it? Thanks!

Yes thanks, merged just now; was waiting for the tests to complete, since we don't have auto merge enabled here.

**Pull request merged**